File: packages/adapters/treesitter/fixtures/sample.go
package ledger

import "fmt"

type Ledger struct {
	total int
}

func (l *Ledger) Post(amount int) {
	l.total += amount
}

func Audit(l *Ledger) bool {
	fmt.Println(l.total)
	return l.total > 0
}

File: packages/adapters/treesitter/fixtures/sample.rs
use std::io::Read;

pub struct Ledger {
    total: u64,
}

impl Ledger {
    pub fn post(&mut self, amount: u64) {
        self.total += amount;
    }
}

pub trait Audit {
    fn check(&self) -> bool;
}

pub fn audit(l: &Ledger) -> bool {
    l.total > 0
}
